### ISSUE TITLE
Ignoring `stopOnFirstError`

### ISSUE THREAD

**Issue by ndench** [vimeo/psalm#506]
Following the [configuration guide](https://github.com/vimeo/psalm/wiki/Configuration) I set the `stopOnFirstError` attribute of my `psalm.xml`, however it doesn't appear to be working.
I get the same output whether I have:

```xml
<psalm
    name="Copied from https://github.com/vimeo/psalm/blob/master/assets/config_levels/5.xml"
    stopOnFirstError="false"
>
```

or

```xml
<psalm
    name="Copied from https://github.com/vimeo/psalm/blob/master/assets/config_levels/5.xml"
    stopOnFirstError="true"
>
```

Additionally, fixing an error reported by psalm then running it again results in a new error a little further on in the code which wasn't reported before. See this example code:

```php
<?php
class Foo
{
  public static function create(): ?Foo
  {
    if (rand(0, 1)) {
      return null;
    }
    
    return new Foo();
  }
  
  public function doFoo(): int
  {
    return rand(0, 100);
  }
}

$foo = Foo::create();
$foo->doFoo();
$foo->doFoo();
```

Running this code on [getpsalm.org](https://getpsalm.org) gives an error for the first call of `$foo->doFoo()` but not the second.

Am I misunderstanding how `stopOnFirstError` works?

**Comment by muglug** [comment]
Sorry - `stopOnFirstError="true"` doesn't work anymore - it didn't work properly for the best part of a year, and I recently removed its support completely (though didn't remove from the wiki, which I've just done).

**Comment by ndench** [comment]
Ah fair enough. It would be a really great feature though, is there another way it could be implemented?

**Comment by muglug** [comment]
It would be possible with single-threaded execution, but not possible with threaded execution (`--threads=X). It's due to that conflict that it stopped working.
